aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/emitters/UDPEmitter.java

```diff
@@ -58,17 +58,20 @@ public boolean sendSubsegment(Subsegment subsegment) {
         if (logger.isDebugEnabled()) {
             logger.debug(subsegment.prettyStreamSerialize());
         }
-        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes());
+        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes(), subsegment);
     }
 
-    private boolean sendData(byte[] data) {
+    private boolean sendData(byte[] data, Entity entity) {
         DatagramPacket packet = new DatagramPacket(sendBuffer, DAEMON_BUF_RECEIVE_SIZE, config.address);
         packet.setData(data);
         try {
             logger.debug("Sending UDP packet.");
             daemonSocket.send(packet);
         } catch (IOException e) {
             logger.error("Exception while sending segment over UDP.", e);
+            if (logger.isInfoEnabled()) {
+                logger.info("Failed to send the following entity: " + entity.prettySerialize());
```

Comment:
Segment/subsegment documentation has `annotation` and `metadata` key which could contain sensitive customer data. Right now the only documentation printing on log file is debug mode so normally no content would be written down in production. This might not suit most use cases of the customers. Any reason not using `DEBUG` here?

Reply:
Is there any other way how we can give a use a pointer which entity was too big?
Because this is something you want to know beyond the debug level I think.
If you enable DEBUG, you see all xray traces passing by, as this trace want to give the end use a clue why it failed.

Reply:
@haotianw465 I understand you security concerns.  I will update the patch to print the segment id in the error trace.